src/sagemaker/model.py

```diff
@@ -352,7 +370,26 @@ def register(
         if model_package_group_name is not None:
             container_def = self.prepare_container_def()
         else:
-            container_def = {"Image": self.image_uri, "ModelDataUrl": self.model_data}
+            container_def = {
+                "Image": self.image_uri,
+                "ModelDataUrl": self.model_data,
+            }
+        if (
+            framework is not None
+            and framework_version is not None
+            and nearest_model_name is not None
+            and data_input_configuration is not None
+        ):
+            container_def.update(
+                {
+                    "Framework": framework,
+                    "FrameworkVersion": framework_version,
+                    "NearestModelName": nearest_model_name,
+                    "ModelInput": {
+                        "DataInputConfig": data_input_configuration,
+                    },
+                }
+            )
```

Comment:
This part of code duplicate in multiple places. Can we extract it to a helper method so it can be reused?

Reply:
migrated the code to a util function